rust/lance-encoding/src/encodings/logical/list.rs

```diff
@@ -679,20 +679,11 @@ impl LogicalPageDecoder for ListPageDecoder {
             self.lists_available = 0;
             let offset_wait_start = self.rows_drained + self.lists_available;
             let item_start = self.offsets[offset_wait_start as usize];
-            let mut items_needed =
+            let items_needed =
                 self.offsets[offset_wait_start as usize + num_rows as usize] - item_start;
             if items_needed > 0 {
-                // First discount any already available items
-                let items_already_available = self.item_decoder.as_mut().unwrap().avail();
-                trace!(
-                    "List's items decoder needs {} items and already has {} items available",
-                    items_needed,
-                    items_already_available,
-                );
-                items_needed = items_needed.saturating_sub(items_already_available);
-                if items_needed > 0 {
-                    self.item_decoder.as_mut().unwrap().wait(items_needed).await?;
-                }
```

Comment:
A long time ago the list decoder had a lot of the same code that is now in the struct decoder.  Now the list decoder actually uses a struct decoder.  When we made that switch we should have yanked this block out.  We don't need to decrement `items_needed` because `self.item_decoder` will already decrement it.  As a result we were double counting.  In most cases this was not an issue.  We rarely double counted by much and the double counting wouldn't have an impact (as long as we `wait` the proper number of physical pages then everything works even if the amount we wait on each page is a little inaccurate.